hare/eligibility/oracle.go

```diff
@@ -69,7 +110,7 @@ type vrfMessage struct {
 
 // buildVRFMessage builds the VRF message used as input for the BLS (msg=Beacon##Id##Layer##Round)
 func (o *Oracle) buildVRFMessage(id types.NodeId, layer types.LayerID, round int32) ([]byte, error) {
-	v, err := o.beacon.Value(safeLayer(layer))
+	v, err := o.beacon.Value(safeLayer(layer, k))
```

Comment:
it is better if you'll pass to the beacon the current layer and the beacon will calculate the safe layer and use it. Using the safe layer is part of the beacon functionality and shouldn't concern the caller of this method.

Reply:
It is a matter of perspective. I see the beacon as just a value provider and the Oracle as the entity that knows the meaning of safe layer.